### PR TITLE
Use poetry for dependency management

### LINKED ISSUE #66: Use poetry for dependency management
[Poetry](https://python-poetry.org/) is a modern way to manage python projects. Right now dependency management is a bit messy with 2 separate dependency lists in `requirements` and `.github/workflows/constraints.txt`. And you need to manually create virtualenv if you don't want to mess with system packages.
- Poetry can install all project dependencies in virtualenv with `poetry install` and it manages dependency versions.
- It separates between normal dependencies and dev dependencies (black, flake8, etc)
- Then you can run any tools with `poetry run`, e.g. `poetry run black`, can be integrated with `pre-commit` as well.
- Some extra configuration is required for `dependabot`.

I want to switch everything (pre-commit config, github actions) to use poetry.
Any objections?

### PR DESCRIPTION
**Pull request by KapJI**
Remove old requirements files and create `pyproject.yaml` with poetry settings.
- Keep `0.0.0` as a placeholder, we're not going to build package for PyPI anyway. Component version should be placed in `manifest.json`.
- Change Python version to 3.8, as it's [current required version for home assistant](https://github.com/home-assistant/core/blob/79b5ca9415a9977f6f030d055149f57645e57d3e/homeassistant/const.py#L7). If you want, we can run lints for both 3.8 and 3.9.

### How to use poetry
- Install Poetry using pipx, pip or homebrew.
- Run `poetry install` to setup environment.
- (optional) Run `poetry run pre-commit install` to setup `pre-commit` git hook.
- Then you can run `poetry run pre-commit run --all-files` or just `git commit` (if you installed git hook).
- New dev dependencies can be added with `poetry add --dev flake8`, normal dependencies still need to be specified in `manifest.json`.

Closes #66

### CODE REVIEW

**Comment by KapJI** [comment]
@leikoilja @DurgNomis-drol please take a look 🙂

File: poetry.lock [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+colorama = ["colorama (>=0.4.3)"]
+d = ["aiohttp (>=3.3.2)", "aiohttp-cors"]
+
+[[package]]
+name = "cached-property"
+version = "1.5.2"
+description = "A decorator for caching properties in classes."
+category = "dev"
+optional = false
+python-versions = "*"
```
Comment (leikoilja) [inline poetry.lock:48]: Do we need it? Or we have a plan to set start utilizing it?

File: poetry.lock [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+
+[[package]]
+name = "nodeenv"
+version = "1.5.0"
+description = "Node.js virtual environment builder"
+category = "dev"
+optional = false
+python-versions = "*"
+
```
Comment (leikoilja) [inline poetry.lock:129]: Also not very sure why we need this one 🤔

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+[tool.poetry]
+name = "ha-google-home"
+version = "0.0.0"
```
Comment (leikoilja) [inline pyproject.toml:3]: Why do we set version to 0.0.0? Will it have to correspond to package version when we do a release?

**Review by leikoilja — APPROVED** [review]
It looks good to me, but a pretty new concept. All the requirements moved from 'requirements' file are automatically migrated to poetry.lock, right? 
There are few packages that I am not sure we need or are using, but I guess we'll just leave them for now 🤷‍♂️😅 
Thanks a lot for making the dev setup much better for us. Will you please also update README in contribution section to add a few lines how one should operate and use poetry now. (Pretty much all you have already mentioned in the issue)

File: poetry.lock (hunk 1 above)
Comment (KapJI) [reply poetry.lock:48]: This is autogenerated file and we shoulnd't edit it by hand. All these packages are installed as dependencies for those are in `pyproject.yaml`.

File: pyproject.toml (hunk 3 above)
Comment (KapJI) [reply pyproject.toml:3]: Otherwise we will need to keep them in sync, it's better when you need to change it only in one place.

Comment (leikoilja) [reply pyproject.toml:3]: That's a good point. I'm wondering I go there is GitHub action or some utility to manage the versioning across many files based on gh release number 😅

Comment (KapJI) [reply pyproject.toml:3]: There is a plugin for poetry which can use git tags: https://github.com/mtkennerly/poetry-dynamic-versioning
But Home Assistant components are not regular python packages, so it's not very useful here. You can try it for `glocaltokens` though.

**Comment by DurgNomis-drol** [comment]
I don't no anything about dependency management, but it looks very easy to use and maintain :smile: 

I just follow along reading what both of you are talking about and learning. It is approved from my perspective :+1: 

Thanks @KapJI for introducing us to it :rocket:

File: pyproject.toml (hunk 3 above)
Comment (leikoilja) [reply pyproject.toml:3]: Oh perfect, that will be a good place to practice! Thanks man 🤩💥

**Pull request merged**